lib/mongo_client.js

```diff
@@ -67,6 +67,7 @@ const closeOperation = require('./operations/mongo_client_ops').closeOperation;
  * @property {string} [kmsProviders.local.key] The master key used to encrypt/decrypt data keys
  * @property {object} [schemaMap] A map of namespaces to a local JSON schema for encryption
  * @property {boolean} [bypassAutoEncryption] Allows the user to bypass auto encryption, maintaining implicit decryption
+ * @param {string} [cryptdConnectionString] A local process the driver communicates with to determine how to encrypt values in a command. Defaults to "mongodb://%2Fvar%2Fmongocryptd.sock" if domain sockets are available or "mongodb://localhost:27020" otherwise.
```

Comment:
Please take another look at the spec I linked you to earlier. I'm indifferent to whether we should have a nested `extraOptions` or put the extra `mongocryptd`-specific options at the top level here, but the documentation for the `mongocryptdURI` is:
```
   // Defaults to "mongodb://%2Fvar%2Fmongocryptd.sock" if domain sockets are available or
   // "mongodb://localhost:27020" otherwise.
   mongocryptdURI: Optional<String>,
```
I think this would seem to indicate that we should do the parsing in the `MongoClient` constructor, and by the time it gets to actually connecting here we should _only_ be using `mongoClient.s.options.autoEncryption.extraOptions.mongocryptdURI`

Reply:
Renamed as `autoEncryption.extraOptions.mongocryptURI`.